src/components/robot/Robot.ts

```diff
@@ -0,0 +1,195 @@
+   *   passing along a proto type.
+   * @group Frame System
+   * @alpha
+   */
+  frameSystemConfig(transform: Transform[]): Promise<proto.FrameSystemConfig[]>;
+
+  /**
+   * Transform a given source Pose from the reference frame to a new specified
+   * destination which is a reference frame.
+   *
+   * @privateRemarks
+   *   TODO: this function should return an more idiomatic type instead of just
+   *   passing along a proto type.
+   * @param query - The pose that should be transformed
+   * @param destination - The name of the reference frame to transform the given
+   * @param supplementalTransforms - Pose information on any additional
+   *   reference frames that are needed to perform the transform
+   * @group Frame System
+   * @alpha
+   */
+  transformPose(
+    source: PoseInFrame,
+    destination: string,
+    supplementalTransforms: Transform[]
+  ): Promise<PoseInFrame>;
+
+  /**
+   * Transform a given source point cloud from the reference frame to a new
+   * specified destination which is a reference frame.
+   *
+   * @param pointCloudPCD - The point clouds to transform. This should be in the
+   *   PCD format encoded into bytes:
+   *   https://pointclouds.org/documentation/tutorials/pcd_file_format.html
+   * @param source - The reference frame of the point cloud.
+   * @param destination - The reference frame into which the source data should
+   *   be transformed, if unset this defaults to the "world" reference frame. Do
+   *   not move the robot between the generation of the initial pointcloud and
+   *   the receipt of the transformed pointcloud because that will make the
+   *   transformations inaccurate.
+   * @group Frame System
+   * @alpha
+   */
+  transformPCD(
+    pointCloudPCD: Uint8Array,
+    source: string,
+    destination: string
+  ): Promise<Uint8Array>;
+
+  /**
+   * Get the list of discovered component configurations.
+   *
+   * @privateRemarks
+   *   TODO: this function should return an more idiomatic type instead of just
+   *   passing along a proto type.
+   * @param queries - The list of component models to discovery.
+   * @group Discovery
+   * @alpha
+   */
+  discoverComponents(
+    queries: proto.DiscoveryQuery[]
+  ): Promise<proto.Discovery[]>;
+
+  /**
+   * Create a new session that expects at least one heartbeat within the
+   * returned window. If the window lapses, any resources that have safety heart
+   * monitored methods, where this session was the last caller on the resource,
+   * will be stopped.
+   *
+   * @privateRemarks
+   *   TODO: this function should return an more idiomatic type instead of just
+   *   passing along a proto type.
+   * @param resume - A session ID that is used to attempt to continue a stream
+   *   after a disconnection event. If the session is not found, a new one will
+   *   be created and returned.
+   * @group Sessions
+   * @alpha
+   */
+  startSession(resume: string): Promise<proto.StartSessionResponse>;
```

Comment:
this session stuff bloats the interface. I'd move it to a session manager interface suitable for end users

Reply:
agreed - it's gone now - only session manager exposes session stuff